graylog2-server/src/main/java/org/graylog/plugins/sidecar/services/EtagService.java

```diff
@@ -62,41 +88,82 @@ public EtagService(SidecarPluginConfiguration pluginConfiguration,
 
     @Subscribe
     public void handleEtagInvalidation(EtagCacheInvalidation event) {
-        if (event.etag().equals("")) {
-            LOG.trace("Invalidating all collector configuration etags");
+        //noinspection rawtypes
+        Cache cache;
+        switch (event.cacheContext()) {
+            case COLLECTOR -> cache = collectorCache;
+            case CONFIGURATION -> cache = configurationCache;
+            case REGISTRATION -> cache = registrationCache;
+            default -> throw new IllegalArgumentException("Unknown cache context");
+        }
```

Comment:
Playing around with the new java language features, sorry 😬 . What do you think about this simplification?

```suggestion
        var cache = switch (event.cacheContext()) {
            case COLLECTOR -> collectorCache;
            case CONFIGURATION -> configurationCache;
            case REGISTRATION -> registrationCache;
        };
```

Reply:
sure, why not :+1: